### PR TITLE
IS-interceptors

### PR DESCRIPTION
**Pull request by dmitrykarmazin**

### CODE REVIEW

File: src/app/auth/store/effects/auth.effects.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export class AuthEffects {
 
             return new AuthenticationSuccessAction({token: user.token});
           }),
-          catchError((error) => {
+          catchError((error: any) => {
```
Comment (McSim182) [inline src/app/auth/store/effects/auth.effects.ts]: error can be typed as (error: Error) please avoid using :any type

File: src/app/auth/store/effects/auth.effects.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export class AuthEffects {
...
```
Comment (McSim182) [inline src/app/auth/store/effects/auth.effects.ts]: same as above

File: src/app/auth/store/effects/auth.effects.ts [hunk 3]
```diff
@@ -58,7 +58,7 @@ export class AuthEffects {
 
       return this.authService.getUser(action['payload'].token).pipe(
         map((user: User) => new GetUserInfoSuccessAction(user)),
-        catchError((error) => of(new GetUserInfoFailAction(error)))
+        catchError((error: any) => of(new GetUserInfoFailAction(error)))
```
Comment (McSim182) [inline src/app/auth/store/effects/auth.effects.ts]: same as above

File: src/app/auth/store/effects/auth.effects.ts [hunk 4]
```diff
@@ -73,14 +73,14 @@ export class AuthEffects {
 
       return this.authService.getUser(action.payload).pipe(
         map((user: User) => new GetUserInfoSuccessAction(user)),
-        catchError(error => of(new GetUserInfoFailAction({error: error})))
+        catchError((error: any) => of(new GetUserInfoFailAction({error: error})))
```
Comment (McSim182) [inline src/app/auth/store/effects/auth.effects.ts]: same as above

File: src/app/auth/store/effects/auth.effects.ts [hunk 5]
```diff
@@ -73,14 +73,14 @@ export class AuthEffects {
...
       );
     })
   );
 
   @Effect() onFail$: Observable<Action> = this.actions$.pipe(
     ofType(authActions.GET_USER_INFO_FAIL, authActions.AUTHENTICATE_ERROR),
-    map((error) => {
+    map((error: any) => {
```
Comment (McSim182) [inline src/app/auth/store/effects/auth.effects.ts]: same as above

File: src/app/shop/containers/product-page/product-page.component.ts [hunk 6]
```diff
@@ -70,5 +68,12 @@ export class ProductPageComponent implements OnInit {
       quantity: this.count
     }));
   }
+  numberOnly(event: KeyboardEvent): any {
```
Comment (McSim182) [inline src/app/shop/containers/product-page/product-page.component.ts]: why not to use `<input type="number" />`  then?

File: src/app/shop/containers/product-page/product-page.component.ts [hunk 7]
```diff
@@ -37,19 +38,16 @@ export class ProductPageComponent implements OnInit {
...
-    // TODO:
+    this.getProduct();
   }
 
   getProduct(): void {
-    this.route.params.subscribe( (params: Params) => {
-      const id: string  = params['id'];
-    });
+    this.id = this.route.snapshot.params.id;
+    // this.product = this.store.select(productById(id));
```
Comment (McSim182) [inline src/app/shop/containers/product-page/product-page.component.ts]: why is that commented? 
back-end is working now, we can really fetch the data by id.

File: src/app/shop/containers/shop-page/shop-page.component.ts [hunk 8]
```diff
@@ -25,9 +25,15 @@ export class ShopPageComponent implements OnInit {
 
     this.viewMode$ = of('grid');
     this.products$ = of([
```
Comment (McSim182) [inline src/app/shop/containers/shop-page/shop-page.component.ts]: this should be refactored according to back-end

File: src/app/shop/containers/product-page/product-page.component.ts (hunk 6 above)
Comment (dmitrykarmazin) [reply src/app/shop/containers/product-page/product-page.component.ts]: For user-friendly reason,we has already buttons, it's necessary to duplicate functional?

**Comment by McSim182** [comment]
Closing because we have a duplicate #20

**Pull request closed without merging**